src/components/AsyncComponents.js

```diff
@@ -0,0 +1,21 @@
+import StringsFrontend from '@/data/strings/strings';
+import SingleBackend from '@/data/strings';
+
+function createAsyncPage(context, pageName) {
+  const StringInstance = new StringsFrontend(context, SingleBackend);
+  return () => ({
+    component: StringInstance.initialize()
+      .then(() => pageName())
+      .then(cmp => ({
+        extends: cmp.default,
+        data() {
+          return { Strings: StringInstance };
+        }
+      })),
+    // loading: LoadingComponent, //TODO
+    // error: ErrorComponent, //TODO
```

Comment:
We need to add the error component here! I'd recommend a simple textual message for most components saying the item failed to load (for now)